pkg/reconciler/clusteringress/resources/secret.go

```diff
@@ -28,12 +28,22 @@ import (
 	corev1listers "k8s.io/client-go/listers/core/v1"
 )
 
-// GetSecrets gets the all of the secrets referenced by the given ClusterIngress, and
+// GetSecretsForClusterIngress gets the all of the secrets referenced by the given ClusterIngress, and
 // returns a map whose key is the a secret namespace/name key and value is pointer of the secret.
-func GetSecrets(ci *v1alpha1.ClusterIngress, secretLister corev1listers.SecretLister) (map[string]*corev1.Secret, error) {
+func GetSecretsForClusterIngress(ci *v1alpha1.ClusterIngress, secretLister corev1listers.SecretLister) (map[string]*corev1.Secret, error) {
+	return getSecrets(&ci.Spec, ci.GetName(), secretLister)
+}
+
+// GetSecretsForIngress gets the all of the secrets referenced by the given Ingress, and
+// returns a map whose key is the a secret namespace/name key and value is pointer of the secret.
+func GetSecretsForIngress(i *v1alpha1.Ingress, secretLister corev1listers.SecretLister) (map[string]*corev1.Secret, error) {
```

Comment:
We could combine these with:

```go
type IngressAccessor interface {
   kmeta.Accessor
   GetSpec() *v1alpha1.IngressSpec
}
```

Reply:
Let me look into it.

Reply:
It turns out that the `name` argument is not used.  I have removed it.